launcher/src/Instance.scala

```diff
@@ -103,16 +106,43 @@ object Instance{
             case _ => ???
           }
         case multiple =>
-          throw new Exception(
-            s"More than one running instance found: ${multiple.map(_.instanceId()).mkString(", ")}. You can delete extra instances in the aws-dev-admin console (double check your region!)."
-          )
+          val instanceOptions = multiple.zipWithIndex.map(t => {
+            val (instance, idx) = t
+            val dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss z").withZone(ZoneId.systemDefault())
+            s"\t[${idx + 1}] ${instance.instanceId()} (${instance.placement().availabilityZone()}) launched at ${dateFormatter.format(instance.launchTime())}"
+          }).mkString("\n")
+          log(s"Multiple Devbox instances found. Which one would you like to keep? (enter 0 to discard all and get a new one)\n" + instanceOptions)
+          val chosen = scala.io.StdIn.readInt() - 1
+          val idsToDelete = (multiple.take(chosen) ++ multiple.drop(chosen + 1)).map(_.instanceId())
+          log(s"Terminating instances: ${idsToDelete.mkString(", ")}")
+          ec2.terminateInstances(TerminateInstancesRequest.builder().instanceIds(idsToDelete:_*).build())
+
+          // Wait for the instances to terminate
+          val timeout = 60.seconds.fromNow
+          while ({
```

Comment:
I see that you picked up the weird `while` form 🙂 I guess it has a little advantage in that you can use the local variable defined in this block in the exit condition, so I won't complain that `do-while` is more understandable. 👍

Reply:
@iuliand-db it's not just me, it seems I've infected Martin Odersky with the weird while loops too 😛 

- http://dotty.epfl.ch/docs/reference/dropped-features/do-while.html

Reply:
I know, but I think he's getting too obsessed with reducing the number of keywords in the language :)